### DISCUSSION TITLE
Update coming for security scans and reviews stages

### DISCUSSION THREAD

**Discussion by wpschaub** [WorkSafeBC-Common-Engineering/Common.Engineering.Discussion#4] [Updates - Templates]
**Heads up**, we will post an update to the blueprints to change the name of the **security scans** stage and the flow of the renamed and the security review stage. @604kev can you add a crisp WHY summary here and perhaps craft a blog post for our technical blog do show the now and future state of the pipeline stages?

I will close this thread as soon as the associated pull requests have been posted.

cc @said-akram

**Comment by wpschaub** [comment]
@604kev https://github.com/WorkSafeBC-Common-Engineering/AzureDevOps.Automation.Pipeline.Templates.v2/pull/4 has been completed and merged to the codebase. Will you add the WHY to this thread?

**Comment by wpschaub** [comment]
@said-akram while we wait for @604kev to confirm the WHY, what do you think of the v2 blueprints?

**Comment by littleswancode** [comment]
Hi all, apologies for the very long delay in commenting on this item.

The changes in Security-related environment layouts and automation flows were due to a few reasons:
- Pre-deploy gates only in v2/YAML-based CD pipelines
- Need for reliable & timely SR work item generation
- Change and modernization in our DevSecOps review workflow and tools

To start, the Security Scans environment was renamed to Security Automation, simply because no scans were being run. This is due to the evolution of our workflow and toolchain - most of our automated security tools run in the CI build process, not in the CD deploy process. Due to the limited capability of DAST tools, we run those manually to get the most value out of them. As a result, there was no need anymore for a "scans" step as no scans were run! At the same time, we have moved towards automating the administrative items of the Security Review process, such as generating work items and reports, for the engineer to used in the reviews. 

As well, because the Security Review Approval had to be on the pre-deploy gate of the SR environment, the work item needs to already be generated and existing in our FIFO queue system, before the SR approval is prompted for. As well, we didn't want to inject SR-specific steps into the SY environment. So, it needs to run in a separate environment prior to the SR approval, hence the location of the Security Scans environment in the process.

The end result is that both the Security Scans environment has to run/pass for the Security Review environment to prompt for approval, and both that and the QA environment must pass to deploy to Staging. This not only guarantees the Security Review has been done and passed, but also guarantees there is an auditable paper trail based on the automatically generated Azure DevOps work item, as a part of our SR queue system.

A future evolution of this is going to be a similar separate automation environment being added between Staging and Production. We have a Secondary Security Approval (SR2) gate that happens before deploying to production, that is now using its own separate work item and queue system. So, there would be great value in automating this and treating it the same from an automation flow perspective to the SR1.

Looking forward to moving forward with a lot more updates here!

**Reply by wpschaub** [reply]
Thank you for the explanation @604kev! I will be synch'ing the changes to the *.Templates.v2 repo this weekend.
